CMakeLists.txt

```diff
@@ -1,10 +1,201 @@
+            message(INFO, "OTIO Defaulting Python install to ${OTIO_RESOLVED_PYTHON_INSTALL_DIR}")
+        endif()
+    endif()
+else()
+    set(OTIO_RESOLVED_PYTHON_INSTALL_DIR "${OTIO_PYTHON_INSTALL_DIR}")
+    set(OTIO_RESOLVED_CXX_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}")
+    set(OTIO_RESOLVED_CXX_DYLIB_INSTALL_DIR "${DOTIO_PYTHON_INSTALL_DIR}/opentimelineio")
+    message(INFO, "OTIO Python installing to ${OTIO_RESOLVED_PYTHON_INSTALL_DIR}")
+endif()
+
+set(CMAKE_INSTALL_INCLUDEDIR "${OTIO_RESOLVED_CXX_INSTALL_DIR}/include")
+
+if(OTIO_SHARED_LIBS)
+    message(INFO, "Building shared libs.")
+    set(OTIO_SHARED_OR_STATIC_LIB "SHARED")
+else()
+    message(INFO, "Building static libs.")
+    set(OTIO_SHARED_OR_STATIC_LIB "STATIC")
+    if (OTIO_PYTHON_INSTALL AND NOT MSVC)
+        # TODO: add explicit visibility decorations for OpenTime and OpentimelineIO to resolve
+        # this issue. For reference, there is discussion here: https://gist.github.com/ax3l/ba17f4bb1edb5885a6bd01f58de4d542
+        message(WARNING, "pybind11 forces visibility flags, used shared libraries instead.")
+    endif()
+endif()
+
+if(OTIO_CXX_INSTALL)
+    message(INFO, "Installing C++ bindings to: ${OTIO_RESOLVED_CXX_INSTALL_DIR}")
+    message(INFO, "Installing C++ dynamic libraries to: ${OTIO_RESOLVED_CXX_DYLIB_INSTALL_DIR}")
+
+    if(OTIO_DEPENDENCIES_INSTALL)
+        message(INFO, "  Installing 'any' and 'nonstd' for C++.. (OTIO_DEPENDENCIES_INSTALL:ON)")
+    else()
+        message(INFO, "  Not installing any and nonstd for C++.. (OTIO_DEPENDENCIES_INSTALL:OFF)")
+    endif()
+else()
+    message(INFO, "Install C++ bindings..... OFF")
+endif()
+
+if(OTIO_PYTHON_INSTALL)
+    message(INFO, "Installing Python bindings to: ${OTIO_RESOLVED_PYTHON_INSTALL_DIR}")
+else()
+    message(INFO, "Install Python bindings.. OFF")
+endif()
+
+#------------------------------------------------------------------------------
+# Global language settings
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+if(MSVC)
+	set(PYBIND11_CPP_STANDARD /std:c++11)
+else()
+	set(PYBIND11_CPP_STANDARD -std=c++11)
+endif()
+
+if(OTIO_CXX_COVERAGE AND NOT MSVC)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --coverage")
+    # this causes cmake to produce file.gcno instead of file.cpp.gcno
+    set(CMAKE_CXX_OUTPUT_EXTENSION_REPLACE 1)
+endif()
+
+if(WIN32)
+    # Windows debug builds for Python require a d in order for the module to
+    # load. This also helps ensure that debug builds in general are matched
+    # to the Microsoft debug CRT.
+    set(OTIO_DEBUG_POSTFIX "d")
+endif()
+
+#------------------------------------------------------------------------------
+# Fetch or refresh submodules if requested
+#
+# fetching submodules does not work in Travis, so override the OTIO_AUTOMATIC_SUBMODULES option
+
+if (OTIO_AUTOMATIC_SUBMODULES AND NOT DEFINED ENV{TRAVIS})
```

Comment:
can you add a TODO note to rename the TRAVIS variable to something else?  We aren't using travis anymore for CI